lib/cn/cn_tree.c

```diff
@@ -1480,6 +1482,14 @@ cn_comp_update_split(
 
             w->cw_split.nodev[RIGHT] = right;
             right->tn_cgen++;
+
+            /* Vary the split size to avoid a node-split thundering herd.
+             * Note that the split size of the right node is disjoint
+             * from the split size of the left node.
+             */
+            right->tn_split_size = (size_t)tree->rp->cn_split_size << 30;
+            right->tn_split_size += (right->tn_nodeid % 3) << 30;
+            right->tn_split_size += 5ul << 30;
```

Comment:
What is the significance of the 3 and 5 GB?

Reply:
The 3 (3GB) is used to vary the split size of the node (based on node ID) to avoid all the nodes wanting to split at the same time.  The 5 (5GB) separates the left from the right such that with a uniform load (like ycsb) they will not likely both need to split at the same time.   Consider splitting such that you keep doubling your nodes, when you hit 32 or so the thundering herd becomes very painful.  It was worse when we could only run one split at a time, but in this patch we can now do up to 8 concurrent splits.  But that can still cause a serious jolt to the system when you need to split 128 or more nodes.   So staggering their split sizes eliminates that problem.  Note that this really only kicks in when the spills are tiny (e.g., uniform load once the node count gets really big...

Reply:
Would be good to have this in a comment near the constants :)

Reply:
Lemme give it a try...

Reply:
Staggering node split size might come back to bite when we need to join nodes.

Reply:
What's your thinking on that?  I can't think of anything off hand..